pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -1139,6 +1176,32 @@ static void handleQueryLimitOverride(PinotQuery pinotQuery, int queryLimit) {
     }
   }
 
+  /**
+   * Rewrites 'DistinctCount' to 'SegmentPartitionDistinctCount' for the given PQL broker request.
+   */
+  @Deprecated
+  @VisibleForTesting
+  static void handleSegmentPartitionedDistinctCountOverride(BrokerRequest brokerRequest,
+      Set<String> segmentPartitionedColumns) {
+    if (segmentPartitionedColumns.isEmpty()) {
+      return;
+    }
+    List<AggregationInfo> aggregationsInfo = brokerRequest.getAggregationsInfo();
+    if (aggregationsInfo != null) {
+      for (AggregationInfo aggregationInfo : aggregationsInfo) {
+        if (StringUtils.remove(aggregationInfo.getAggregationType(), '_')
+            .equalsIgnoreCase(AggregationFunctionType.DISTINCTCOUNT.name())) {
+          for (String expr : aggregationInfo.getExpressions()) {
```

Comment:
Suggest checking if there is a single identifier to avoid unexpected rewrite
```suggestion
          List<String> expressions = aggregationInfo.getExpressions();
          if (expressions.size() == 1 && segmentPartitionedColumns.contains(expressions.get(0))) {
            aggregationInfo.setAggregationType(AggregationFunctionType.SEGMENTPARTITIONEDDISTINCTCOUNT.name());
          }
```